cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/KafkaCruiseControlServlet.java

```diff
@@ -305,19 +340,38 @@ private void getClusterLoad(HttpServletRequest request, HttpServletResponse resp
       ClusterModel.BrokerStats brokerStats = _kafkaCruiseControl.cachedBrokerLoadStats();
       String brokerLoad;
       if (brokerStats != null) {
-        brokerLoad = brokerStats.toString();
+        if (json) {
+          brokerLoad = brokerStats.getJSONString();
+        } else {
+          brokerLoad = brokerStats.toString();
+        }
       } else {
         ClusterModel clusterModel = _kafkaCruiseControl.clusterModel(time, requirements);
-        brokerLoad = clusterModel.brokerStats().toString();
+        if (json) {
+          brokerLoad = clusterModel.brokerStatsJSON();
+        } else {
+          brokerLoad = clusterModel.brokerStats().toString();
+        }
+      }
+      if (json) {
+        setJSONResponseCode(request, response, HttpServletResponse.SC_OK);
+      } else {
+        setResponseCode(response, HttpServletResponse.SC_OK);
       }
-      setResponseCode(response, HttpServletResponse.SC_OK);
       response.setContentLength(brokerLoad.length());
       response.getOutputStream().write(brokerLoad.getBytes(StandardCharsets.UTF_8));
     } else if (granularity.toLowerCase().equals(GRANULARITY_REPLICA)) {
       ClusterModel clusterModel = _kafkaCruiseControl.clusterModel(time, requirements);
-      setResponseCode(response, HttpServletResponse.SC_OK);
-      // Write to stream to avoid expensive toString() call.
-      clusterModel.writeTo(response.getOutputStream());
+      if (json) {
+        String data = clusterModel.getJSONString();
```

Comment:
This string construction used to take more than 2 min. Can you run this against a big cluster and see how long does that take?

Reply:
changing the current implementation to a streaming one requires major rewrite of the code, this i'll be doing as a second update of this later.